mongo_orchestration/daemon.py

```diff
@@ -169,9 +171,12 @@ def stop(self):
             if os.path.exists(self.pidfile):
                 os.remove(self.pidfile)
         else:
+            process = psutil.Process(pid)
             # Try killing the daemon process
             try:
-                os.kill(pid, SIGTERM)
+                process.send_signal(SIGTERM)
+                while process.is_running():
+                    time.sleep(0.25)
```

Comment:
What about using  `os.kill(pid, 0)` to check if a process still exists instead of adding the  `psutil` dependency?

Reply:
done